File: src/main/kotlin/com/tollwood/jpa/Employee.kt
package com.tollwood.jpa

import org.jetbrains.annotations.NotNull
import javax.persistence.*


@Entity(name = "EMPLOYEE")
data class Employee(
        @Id
        @GeneratedValue(strategy = GenerationType.AUTO)
        override val id: Long,
        val firstName: String,
        val lastName: String,
        @Embedded
        val address: Address,
        val taxIdent: String,
        @NotNull
        val technicianId: String?,

        @NotNull
        val email: String,
        @NotNull
        val phone: String,
        @Embedded
        val bankDetails: BankDetails
) : BaseEntity()

File: client/src/employees/Employee.ts
import { Address } from "../common/Address";
import BankDetails from "./BankDetails";

export default class Employee {
  id?: number;
  firstName: string = "";
  lastName: string = "";
  taxIdent: string = "";
  email: string = "";
  phone: string = "";
  technicianId: string = "";
  address: Address = new Address();
  bankDetails: BankDetails = new BankDetails();
}
